src/config/constants/priceHelperLps.ts

```diff
@@ -0,0 +1,20 @@
+import tokens from './tokens'
+import { FarmConfig } from './types'
+
+const priceHelperLps: FarmConfig[] = [
+  /**
+   * These LPs are just used to help with price calculation for MasterChef LPs.
+   */
+  {
+    pid: null,
+    lpSymbol: 'QSD-BNB LP',
+    lpAddresses: {
+      97: '',
+      56: '0x7b3ae32eE8C532016f3E31C8941D937c59e055B9',
+    },
+    token: tokens.qsd,
+    quoteToken: tokens.wbnb,
+  },
```

Comment:
Can you go into a bit more detail? Why are they this shape? Suppose to have the same shape as a farm?

Reply:
Yea sure - QSD doesn’t have a BNB or BUSD farm and so we can’t, with the methodology in fetchFarmsPrices, figure out its price.

By adding a QSD-BNB farm, it can figure out its price. In order to do that we need data about QSD’s price relative to BNB though, and so we need to effectively treat this extra LP like a normal farm and fetch details about the LP’s base & quote tokens, importantly tokenPriceVsQuote.

So once the LP helper farm has got all of the smart contract info, it’s passed to the price-fetching function and is used as part of that logic to help the KUN-QSD farm figure out its prices.

Finally, any LPs without pids are filtered out, so that only our masterchef farms (now all with prices) are actually stored to redux.

I just tidied up the flow here, might help understanding as well? https://github.com/pancakeswap/pancake-frontend/blob/655a6c9bc6c047a5fb4bccd1dbfa3a4b21feedf9/src/state/farms/index.ts#L64-L71